service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderController.java

```diff
@@ -131,6 +137,34 @@ public AboutToStartOrSubmitCallbackResponse handlePopulateSelectorMidEvent(
             .build();
     }
 
+    @PostMapping("/populate-care-orders-selector/mid-event")
+    public AboutToStartOrSubmitCallbackResponse populateCareOrders(@RequestBody CallbackRequest callbackRequest) {
+        final CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        final CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+        final OrderTypeAndDocument currentOrder = caseData.getOrderTypeAndDocument();
+        final List<String> errors = new ArrayList<>();
+
+        if (DISCHARGE_OF_CARE_ORDER == currentOrder.getType()) {
+            List<GeneratedOrder> careOrders = dischargeCareOrder.getCareOrders(caseData);
+
+            if (careOrders.isEmpty()) {
+                errors.add("No care orders to be discharged");
```

Comment:
how about we make this a ternary within `.data(...)`? This is so we have only an `if...else if` and also reduce the chance of a congnitive complexity given parent `if` and should in case more things need to be done within either `ifs...else if`